Feat : sendSeen

sahalMoidu opened the pull request.

sahalMoidu committed:
Merge pull request #1 from pedroslopez/master

Implemented changes mentioned in SimiPrambos PR#28 (#52)

sahalMoidu committed:
Merge remote-tracking branch 'upstream/master'

sahalMoidu committed:
feat : sendSeen

pedroslopez commented:
I don't think we should mix sending a message with marking a conversation as seen. Maybe separating sendSeen to a function in the Chat model would be more flexible. What do you think?

sahalMoidu commented:
If you think that is better we can do it that way. I added it in the send message as we were replying to messages without seeing it.

sahalMoidu commented:
@pedroslopez  Would the chat model or the message model be the best place?

pedroslopez commented:
> @pedroslopez Would the chat model or the message model be the best place?

@sahalMoidu I would say the chat, because it's the _chat_ that you mark as seen.

sahalMoidu committed:
Move sendSeen to Chat Model

pedroslopez committed:
auto send seen when a message is sent

If this is not the intended behavior, `sendSeen: false` can be set as part of the options object

sahalMoidu commented on src/util/Injected.js:
```diff
@@ -33,6 +34,17 @@ exports.LoadUtils = () => {
+    window.WWebJS.sendSeen = async (chatId) => {
+        let chat = window.Store.Chat.get(chatId);
+        if (chat !== undefined) {
+            if (chat.getLastMsgKeyForAction === undefined)
```
It is a workaround. getLastMsgKeyForAction() is called internally inside the  sendseen. And in my test this functions were undefined and hence throwing error.

pedroslopez replied:
Do you know when the function would be undefined? When I was taking a look it seemed like it was defined.

sahalMoidu replied:
Almost all the times I had it undefined. I didn’t figure out the real cause. 
Could you run some tests

pedroslopez replied:
I've been testing and I still have not experienced this issue... Can you try again repeating your same steps for testing and verifying if it does occur?

pedroslopez replied:
I'm removing this for now. If an issue occurs later we need to identify when this occurs and introduce this back in if necessary.

pedroslopez committed:
fix: pass sendSeen arg

pedroslopez committed:
remove getLastMsgKeyForAction workaround

pedroslopez committed:
Merge branch 'master' into feat-sendSeen
